Kiosk/App/CardHandler.swift

```diff
@@ -1,107 +1,167 @@
+    func transaction(_ transaction: CFTTransaction, didReceive cardReaderEvent: CFTCardReaderEvent, cardReaderInfo: CFTCardReaderInfo?) {
+        _cardStatus.onNext(cardReaderEvent.statusMessage)
+    }
 
-    func readerIsAttached() {
-        _cardStatus.onNext("Reader is attatched");
+    func transaction(_ transaction: CFTTransaction, didUpdate cardReaderArray: [CFTCardReaderInfo]) {
+        self.cardReader = cardReaderArray.first
+        _cardStatus.onNext("Received new card reader availability, number of readers: \(cardReaderArray.count)")
     }
 
-    func readerIsConnecting() {
-        _cardStatus.onNext("Reader is connecting");
+    func transaction(_ transaction: CFTTransaction, didRequestProcessOption cardInfo: CFTCardInfo) {
+        logger.log("Received request for processing option, will process transaction.")
+        _cardStatus.onNext("Request for process option, automatically processing...")
+        transaction.select(processOption: .process)
+        // TODO: CardFlight docs says we're supposed to confirm with the user before proceeding with the transaction,
+        // but that's silly because we're only tokenizing and not not making an authorization or a charge.
     }
 
-    func readerIsDisconnected() {
-        _cardStatus.onNext("Reader is disconnected");
-        logger.log("Card Reader Disconnected")
+    func transaction(_ transaction: CFTTransaction, didRequestDisplay message: CFTMessage) {
+        let message = message.primary ?? message.secondary ?? "Unknown message"
+        logger.log("Received request to display message: \(message)")
+        _cardStatus.onNext("Received message for user: \(message)")
+        // TODO: Present message to user somehow
+        // TODO: We want to show the user the custom messages but not messages confirming the transaction (see above,
+        // it's not appropriate for tokenization). So we need to show the user _most_ messages but not messages
+        // confirming transactions (that don't exist).
     }
+}
 
-    func readerSwipeDidCancel() {
-        _cardStatus.onNext("Reader did cancel");
-        logger.log("Card Reader was Cancelled")
+typealias UnhandledDelegateCallbacks = CardHandler
+/// We don't expect any of these functions to be called, but they are required for the delegate protocol.
+extension UnhandledDelegateCallbacks {
+    func transaction(_ transaction: CFTTransaction, didDefer transactionData: Data) {
+        logger.log("Transaction has been deferred.")
+        _cardStatus.onNext("Transaction deferred")
     }
 
-    func readerGenericResponse(_ cardData: String!) {
-        _cardStatus.onNext("Reader received non-card data: \(cardData ?? "") ");
-        reader.beginSwipe()
+    public func transaction(_ transaction: CFTTransaction, didRequest cvm: CFTCVM) {
+        logger.log("Transaction requested signature from user, ignoring.")
+        _cardStatus.onNext("Ignoring user signature request from CardFlight")
```

Comment:
This can be a bit more nuanced, since we can determine if a signature is _actually_ required through the `cvm` parameter, described here: https://github.com/CardFlight/cardflight-v4-ios/issues/3